lib/SILGen/ArgumentSource.h

```diff
@@ -222,8 +222,15 @@ class ArgumentSource {
     return Storage.get<LValueStorage>(StoredKind).Loc;
   }
 
+  /// The kind of operation under which we are querying a storage reference.
+  enum class StorageReferenceKind {
```

Comment:
I find this to be pretty misleading as a name. Can you rename it to something with the name operation. Maybe: StorageReferenceOperationKind?

Otherwise, it sounds like you are talking about the actual kind of the StorageReference... which is not what on the face of it you are trying to talk about.